src/pages/Main/index.tsx

```diff
@@ -0,0 +1,75 @@
+import { useState } from "react";
+import ResultBox from "./ResultBox";
+
+import searchImg from "/src/assets/search-20x20.png";
+
+const Menu = () => {
+  const [searchResults, setSearchResults] =
+    useState<kakao.maps.services.PlacesSearchResult>([]);
+
+  const [keyword, setKeyword] = useState("");
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setKeyword(e.target.value);
+  };
+
+  const detectEnter = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.key === "Enter") {
+      searchKeyword();
+    }
+  };
+
+  const searchKeyword = () => {
+    const places = new kakao.maps.services.Places();
+    places.keywordSearch(keyword, (result, status) => {
+      if (status == kakao.maps.services.Status.OK) {
+        setSearchResults(result.slice(0, 3));
+        console.log(result.slice(0, 3));
+      }
+    });
+  };
+
+  return (
+    <div className="vbox pack ml(30) mr(30)">
+      <div className="vbox w(370)">
+        <span className="font(32) GmarketSans 900 c(#363636) dark:c(#fff)">
+          총 1,214개의
+          <br /> 군인 할인점들이 있어요.
+        </span>
+        <span className="font(16) GmarketSans 900 c(#999999)">
+          23.09.19 업데이트
+        </span>
+      </div>
+      <div className="wrap w(370) mt(50) mb(50)">
+        <span className="font(16) GmarketSans 900 c(#363636) dark:c(#fff) mb(10)">
+          어디서 찾아볼까요?
+        </span>
+        <div className="w(370) h(40) b(2/#505050) r(20) pl(12) hbox">
+          <img
+            src={searchImg}
+            className="w(20) h(20) mr(12) pointer"
+            alt="search"
+            onClick={searchKeyword}
+          />
+          <input
+            onChange={onChange}
+            onKeyDown={detectEnter}
+            value={keyword}
+            className="placeholder::font(14)+900+c(#999999)+NanumGothic NanumGothic font(14) 900 c(#363636) dark:c(#fff) w(300) dark:bg(#000.0)"
+            placeholder="강남구 삼성동"
+          />
+        </div>
+        <ResultBox result={searchResults} />
```

Comment:
검색을 안했을 때와 검색 결과가 없을 때를 분류하기 위해 위에서 `searchResult` 의 기본값을 `null`으로 주고 여기에서는 null 이 아닐때만 보여주는 게 어떨까요?

```suggestion
        { searchResults && <ResultBox result={searchResults} /> }
```

Reply:
이건 위의 리뷰 반영을 하면서 함께 작업될 것 같습니다!